openmdao/core/group.py

```diff
@@ -783,6 +783,29 @@ def _setup(self, comm, prob_meta):
         # determine which connections are managed by which group, and check validity of connections
         self._setup_connections()
```

Comment:
OpenMDAO [allows connections to be issued in configure](https://openmdao.org/newdocs/versions/latest/theory_manual/setup_stack.html#usage-of-setup-vs-configure), so this check should happen at the end of configure.